src/landscape.cpp

```diff
@@ -1378,11 +1378,9 @@ static std::tuple<bool, bool> FlowRiver(TileIndex spring, TileIndex begin, uint
 		std::tie(found, main_river) = FlowRiver(spring, end, min_river_length);
 	} else if (count > 32) {
 		/* Maybe we can make a lake. Find the Nth of the considered tiles. */
-		TileIndex lakeCenter = 0;
-		int i = RandomRange(count - 1) + 1;
-		std::set<TileIndex>::const_iterator cit = marks.begin();
-		while (--i) cit++;
-		lakeCenter = *cit;
+		std::set<TileIndex>::const_iterator cit = marks.cbegin();
+		std::advance(cit, RandomRange(count - 1) + 1);
```

Comment:
Not sure if this `+ 1` is correct as the old loop was decremented it `--i`

Reply:
RandomRange(count - 1) => [0..(count-2)]
i => [1..(count-1)]
while => up to (count-2) increments (if I'm not wrong)

So I'd say the `+ 1` in new code is wrong.